src/DistributedLock.Etcd/EtcdLeaseDistributedLock.cs

```diff
@@ -0,0 +1,198 @@
+/// </summary>
+public sealed partial class EtcdLeaseDistributedLock : IInternalDistributedLock<EtcdLeaseDistributedLockHandle>
+{
+    private readonly EtcdClientWrapper _etcdClient;
+
+    private readonly (TimeoutValue duration, TimeoutValue renewalCadence, TimeoutValue minBusyWaitSleepTime,
+        TimeoutValue maxBusyWaitSleepTime) _options;
+
+    public EtcdLeaseDistributedLock(IEtcdClient client, string lockName, Action<EtcdLeaseOptionsBuilder>? options = null)
+    {
+        this.LockName = lockName ?? throw new ArgumentNullException(nameof(lockName));
+        if (lockName.Length == 0) { throw new FormatException($"{nameof(lockName)}: may not have an empty file name"); }
+
+        this._etcdClient = new EtcdClientWrapper(client);
+        this._options = EtcdLeaseOptionsBuilder.GetOptions(options);
+    }
+
+    // todo revisit API
+    /// <summary>
+    /// Implements <see cref="IDistributedLock.Name"/>
+    /// </summary>
+    public string LockName { get; }
+
+
+    ValueTask<EtcdLeaseDistributedLockHandle?> IInternalDistributedLock<EtcdLeaseDistributedLockHandle>.
+        InternalTryAcquireAsync(TimeoutValue timeout, CancellationToken cancellationToken) =>
+        BusyWaitHelper.WaitAsync(
+            state: this,
+            tryGetValue: (@this, token) => @this.TryAcquireAsync(token),
+            timeout: timeout,
+            minSleepTime: this._options.minBusyWaitSleepTime,
+            maxSleepTime: this._options.maxBusyWaitSleepTime,
+            cancellationToken
+        );
+
+    private async ValueTask<EtcdLeaseDistributedLockHandle?> TryAcquireAsync(CancellationToken cancellationToken)
+    {
+        cancellationToken.ThrowIfCancellationRequested();
+        // 1 lease per application is enough in my case
+        var leaseResponse = await this._etcdClient.LeaseGrantAsync(
+            new LeaseGrantRequest { TTL = this._options.renewalCadence.InSeconds },
+            cancellationToken: cancellationToken);
+        var leaseId = leaseResponse.ID;
+        var cancellationTokenSource = new CancellationTokenSource();
+        // this will start a task to renew the lease every 1/3 of the ttl, should i use the ILeaseHandle here instead
+        _ = this._etcdClient.LeaseKeepAlive(leaseId, cancellationTokenSource.Token);
+        var response =
+            await this._etcdClient.LockAsync(
+                new LockRequest { Name = Google.Protobuf.ByteString.CopyFromUtf8(this.LockName), Lease = leaseId,  },
+                cancellationToken: cancellationTokenSource.Token).ConfigureAwait(false);
+        var actualKey = response.Key;
+        return new EtcdLeaseDistributedLockHandle(actualKey.ToString(), this._etcdClient, leaseId);
+    }
+
+    public string Name => this.LockName;
+
+    public static string GetSafeName(string name)
+    {
+        // TODO figure
+        return DistributedLockHelpers.ToSafeName(name, 1000, s => s);
+    }
+}
+
+/// <summary>
+/// basically similar to AzureBlobLeaseDistributedLock
```

Comment:
Doc comments are end-user-facing: let's update this to something more user-friendly